pkg/operator/cluster.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Some of the code below came from https://github.com/coreos/etcd-operator
+which also has the apache 2.0 license.
+*/
+package operator
+
+import (
+	"fmt"
+
+	"github.com/rook/rook/pkg/cephmgr/client"
+	cephmon "github.com/rook/rook/pkg/cephmgr/mon"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/api"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"github.com/rook/rook/pkg/operator/mon"
+	"github.com/rook/rook/pkg/operator/osd"
+	"github.com/rook/rook/pkg/operator/rgw"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+type Cluster struct {
+	Namespace string
+	factory   client.ConnectionFactory
+	clientset *kubernetes.Clientset
+	Metadata  v1.ObjectMeta `json:"metadata,omitempty"`
+	Spec      ClusterSpec   `json:"spec"`
+}
+
+type ClusterSpec struct {
+	// Version is the expected version of the rook container to run in the cluster.
+	// The rook-operator will eventually make the rook cluster version
+	// equal to the expected version.
+	Version string `json:"version"`
+
+	// Paused is to pause the control of the operator for the rook cluster.
+	Paused bool `json:"paused,omitempty"`
```

Comment:
how would a k8s user toggle this field?  would this be in the `kubectl update` path?